boilerplate/app/navigators/navigation-utilities.tsx

```diff
@@ -83,17 +62,23 @@ export function useBackButtonHandler(
 
     // Unsubscribe when we're done
     return () => BackHandler.removeEventListener("hardwareBackPress", onBackPress)
-  }, [ref])
+  }, [])
 }
 
 /**
  * Custom hook for persisting navigation state.
  */
 export function useNavigationPersistence(storage: any, persistenceKey: string) {
   const [initialNavigationState, setInitialNavigationState] = useState()
-  const [isRestoringNavigationState, setIsRestoringNavigationState] = useState(true)
 
-  const routeNameRef = useRef()
+  // This feature is particularly useful in development mode.
+  // It is selectively enabled in development mode with
+  // the following approach. If you'd like to use navigation persistence
+  // in production, remove the __DEV__ and set the state to true
+  const [isRestored, setIsRestored] = useState(__DEV__ ? false : true)
```

Comment:
```suggestion
  const [isRestored, setIsRestored] = useState(!__DEV__)
```

Linter prefers this -- so let's do that.